tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/RepositoryScriptService.cs

```diff
@@ -0,0 +1,164 @@
+using System.Collections.Specialized;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Services;
+
+public interface IRepositoryScriptService
+{
+    Task<string> GetCommand(string commandName, string packagePath, CancellationToken ct);
+    Task<bool> HasImplementation(string commandName, string packagePath, CancellationToken ct);
+    Task<(bool invoked, ProcessResult result)> TryInvoke(
+        string commandName,
+        string packagePath,
+        OrderedDictionary args,
+        bool invokeFromRepoRoot = true,
+        CancellationToken ct = default
+    );
+    Task<(bool invoked, ProcessResult result)> TryInvoke(string commandName, string packagePath, OrderedDictionary args, CancellationToken ct = default);
+}
+
+public class RepoCommandContract
+{
+    [JsonPropertyName("tags")]
+    public List<string> Tags { get; set; } = [];
+
+    [JsonPropertyName("command")]
+    public string Command { get; set; } = string.Empty;
+}
+
+/// <summary>
+/// Provides repository-specific script discovery and execution by loading the automation config,
+/// caching command mappings per package path, and invoking the associated PowerShell scripts.
+/// </summary>
+/// <param name="logger">The logger used to record informational and warning messages.</param>
+/// <param name="gitHelper">Helper that locates the repository root for a given package path.</param>
+/// <param name="powershellHelper">Helper responsible for running PowerShell commands.</param>
+public class RepositoryScriptService(
+    ILogger<RepositoryScriptService> logger,
+    IGitHelper gitHelper,
+    IPowershellHelper powershellHelper
+) : IRepositoryScriptService
+{
+    public string ScriptConfig = Path.Join("eng", "azsdk-cli-command-overrides.json");
+
+    private readonly Dictionary<string, Dictionary<string, string>> commandCache = [];
+
+    private readonly JsonSerializerOptions jsonOptions = new()
+    {
+        WriteIndented = false,
+        ReadCommentHandling = JsonCommentHandling.Skip,
+        AllowTrailingCommas = true
+    };
+
+    private async Task Load(string repoRoot, CancellationToken ct)
+    {
+        if (commandCache.TryGetValue(repoRoot, out var _))
```

Comment:
The cache might be invalidated when the local repo switches to a different branch.